packages/editor/src/components/post-card-panel/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Icon,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	PanelBody,
+} from '@wordpress/components';
+import { page as pageIcon, navigation, symbol } from '@wordpress/icons';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { humanTimeDiff } from '@wordpress/date';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+const CARD_ICONS = {
+	wp_block: symbol,
+	wp_navigation: navigation,
+};
+
+export default function PostCardPanel( { className, actions, children } ) {
+	const { modified, title, type, templateInfo } = useSelect( ( select ) => {
+		const {
+			getEditedPostAttribute,
+			getCurrentPostType,
+			getCurrentPostId,
+			__experimentalGetTemplateInfo,
+		} = select( editorStore );
+		const { getEditedEntityRecord } = select( coreStore );
+		const _type = getCurrentPostType();
+		const _id = getCurrentPostId();
+		let _templateInfo;
+		if ( _type === 'wp_template' || _type === 'wp_template_part' ) {
+			const _record = getEditedEntityRecord( 'postType', _type, _id );
+			_templateInfo = __experimentalGetTemplateInfo( _record );
+		}
+		return {
+			title: _templateInfo?.title || getEditedPostAttribute( 'title' ),
+			modified: getEditedPostAttribute( 'modified' ),
+			id: _id,
+			type: _type,
+			templateInfo: _templateInfo,
+		};
+	} );
+	const description =
+		type === 'wp_template' || type === 'wp_template_part'
+			? templateInfo?.description
+			: sprintf(
+					// translators: %s: Human-readable time difference, e.g. "2 days ago".
+					__( 'Last edited %s' ),
+					humanTimeDiff( modified )
+			  );
```

Comment:
This probably will close #39312. WDYT?

Reply:
Should we also show the `last edited` for templates too? 🤔

Reply:
I added the last edited to the templates too.

Reply:
This is how it looks now with description and last edited date: 

<img width="284" alt="Screenshot 2024-03-20 at 9 33 30 AM" src="https://github.com/WordPress/gutenberg/assets/16275880/adc2d08a-cca4-4527-b64b-a3c47b1df262">

@jameskoster do we need any polishing here?

Reply:
Thanks for the ping. I pushed a couple of changes to match the latest styles on trunk.